flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/AbstractBinaryWriter.java

```diff
@@ -181,6 +181,29 @@ public void writeDecimal(int pos, Decimal value, int precision) {
 		}
 	}
 
+	@Override
+	public void writeTimestamp(int pos, SqlTimestamp value, int precision) {
+		if (SqlTimestamp.isCompact(precision)) {
+			assert 0 == value.getNanoOfMillisecond();
+			writeLong(pos, value.getMillisecond());
+		} else {
+			// store the nanoOfMillisecond in fixed-length part as offset and nanoOfMillisecond
+			ensureCapacity(8);
+
+			if (value == null) {
```

Comment:
Why we checking null in non compact case. I think this should be consistent with compact branch? Either we check null in both side or we don't check null both.

Reply:
For the fixed-length objects, the writer calls setNullAt() to update fixed-length part（which set null bits and initialize fixed-length part with 0）; For the variable-length objects, the writer calls setNullAt to update fixed-length part and need to assign & initialize variable-length part.